services/cd-service/pkg/valid/validations.go

```diff
@@ -17,23 +17,39 @@ Copyright 2023 freiheit.com*/
 package valid
 
 import (
+	"os"
 	"regexp"
+	"strconv"
 	"strings"
 )
 
 const (
-	MaxAppNameLen  = 39
-	AppNameRegExp  = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
-	TeamNameRegExp = AppNameRegExp
-	EnvNameRegExp  = AppNameRegExp
+	FallbackMaxAppNameLength      = 39
+	KUBERPULT_MAX_APP_NAME_LENGTH = "KUBERPULT_MAX_APP_NAME_LENGTH"
+	AppNameRegExp                 = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
+	TeamNameRegExp                = AppNameRegExp
+	EnvNameRegExp                 = AppNameRegExp
 )
 
 var (
 	applicationNameRx = regexp.MustCompile(AppNameRegExp)
 	teamNameRx        = regexp.MustCompile(TeamNameRegExp)
 	envNameRx         = regexp.MustCompile(EnvNameRegExp)
+	MaxAppNameLen     = setupMaxAppNameLen()
 )
 
+func setupMaxAppNameLen() int {
+	maxAppNameLength := FallbackMaxAppNameLength
+	res, ok := os.LookupEnv(KUBERPULT_MAX_APP_NAME_LENGTH)
+	if ok {
```

Comment:
to change it to `boolean` (see comment above), we also need to adapt this and some other places.
```suggestion
	res, ok := os.LookupEnv(KUBERPULT_MAX_APP_NAME_LENGTH)
	if ok && res == "true" {
```